### DISCUSSION TITLE
[Routing] Route annotation error

### DISCUSSION THREAD

**Discussion by szal1k** [symfony/symfony#52410] [Q&A]
### Symfony version(s) affected

6.3.5

### Description

We have controller with Route annotations like this:
```php
#[Route('/job-carrier')]
class JobCarrierController extends BaseController
{
    #[Route('/{id}/{context}', name: 'job_carrier', methods: ['GET', 'POST'])]
    public function carrier() {}

    #[Route('/_cancel/{id}/{context}', name: 'job_carrier_cancel', methods: ['GET', 'POST'])]
    public function carrierCancel() {}
}
```

When we will create url to "carrierCancel()" method, get 404 not found error. In response i can see cause of the error:

```php
_route_params ["context" => "95"   "id" => "_cancel" ]
```

EntityValueResolver says not found because they get "_cancel" (part of my route), as "id".

### How to reproduce

https://github.com/szal1k/symfony_issue_52392

**Comment by n0rbyt3** [comment]
```php
#[Route('/_cancel/{id}/{context}', name: 'job_carrier', methods: ['GET', 'POST'])]
public function carrierCancel() {}
```
Have a closer look at your definition. You have defined the route name "job_carrier" twice.

Also your reproducer does not create any route or redirect programmatically. Did I miss something?

**Comment by szal1k** [comment]
> ```
> #[Route('/_cancel/{id}/{context}', name: 'job_carrier', methods: ['GET', 'POST'])]
> public function carrierCancel() {}
> ```
> 
> Have a closer look at your definition. You have defined the route name "job_carrier" twice.
> 
> Also your reproducer does not create any route or redirect programmatically. Did I miss something?

Thanks for your response. It is mistake in my description, second route name is "job_carrier_cancel" and not works. Please check reproduce repo.

**Comment by n0rbyt3** [comment]
Got your problem. If you're opening ```/_cancel/95```, the first route ```app_test``` matches, since ```id``` has no requirements. If ```id``` is always a number, add ```requirements: {id: "\d+"}``` to your routes.

**Comment by gordinskiy** [comment]
In many cases this error can be avoided by specifying the identifier format: 
```php
#[Route('/_cancel/{id}/{context}', name: 'job_carrier', methods: ['GET', 'POST'], requirements: ['id' => '\d+'])]
public function carrier() {}
```

**Comment by DevNwamini** [comment]
#[Route('/{id}/{context}', name: 'job_carrier', methods: ['GET', 'POST'])] 
public function carrier() {}

#[Route('/_cancel/{id}/{context}', name: 'job_carrier_cancel', methods: ['GET', 'POST'])] 
public function carrierCancel() {}

 carrier() and carrierCancel() actions have the same name 'job_carrier', To fix this issue, you should give each route a unique name..

**Comment by ghost** [comment]
It's `attribute`, not `annotation`. 
```php
#[SomeAttribute]

/**
 * @SomeAnnotation
 */ 
```

**Comment by tacman** [comment]
add a priority: property to the routes, or reorder them. _cancel is being treated as the id
